cmd/gh/main.go

```diff
@@ -51,7 +51,11 @@ func main() {
 	}
 
 	cmd, _, err := rootCmd.Traverse(expandedArgs)
-	if err != nil || cmd == rootCmd {
+	if err != nil {
+		fmt.Fprintf(stderr, "failed to parse the command:  %s\n", err)
+		os.Exit(2)
```

Comment:
Since the cause of the crash is that we're passing a nil pointer to `IsAuthCheckEnabled()`, could you remove this condition and instead check that `cmd` is non-nil before invoking `IsAuthCheckEnabled`?